src/app/components/planner-list/planner-list.module.ts

```diff
@@ -61,11 +60,11 @@ import {
 import { IterationState, initialState as initialIterationState } from './../../states/iteration.state';
 import { iterationReducer } from './../../reducers/iteration-reducer';
 
+export function httpfactory(backend: XHRBackend, options: RequestOptions, auth: AuthenticationService){
```

Comment:
This is necessary otherwise transpilation fails with 
```
Error: /home/ijarif/Projects/fabric8-planner/src/app/components/work-item-comment/work-item-comment.module.ts:35:1: Error encountered in metadata generated for exported symbol 'WorkItemCommentModule': 
 /home/ijarif/Projects/fabric8-planner/src/app/components/work-item-comment/work-item-comment.module.ts:22:5: Metadata collected contains an error that will be reported at runtime: Reference to a local (non-exported) symbol 'providers'. Consider exporting the symbol.
  {"__symbolic":"error","message":"Reference to a local symbol","line":21,"character":4,"context":{"name":"providers"}}
```

Reply:
@vikram-raj @sanbornsen @sahil143 Do you know a better way of doing this ?
This function is repeated in all the modules that use the `httpService`

Reply:
What you can do is write this block of code at one place and import it from there.